contracts/Vault1.sol

```diff
@@ -1,6 +1,53 @@
-//SPDX-License-Identifier: Unlicense
+//SPDX-License-Identifier: <SPDX-License>
+
 pragma solidity ^0.8.4;
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+contract VaultToken is ERC20 {
+    constructor() ERC20("VAULT", "Vault") {
+        _mint(msg.sender, 100 * 10**uint256(decimals()));
+    }
+}
 
 contract Vault1 {
-    
+    address public owner = msg.sender;
+    VaultToken vaultToken = VaultToken(owner);
+
+    mapping(address => mapping(ERC20 => uint256)) vaultBalance;
```

Comment:
Since in this case vaultToken is fixed, why do we need this internal mapping from ERC20 to uint?